src/components/Tokens/TokenTable/TokenTable.tsx

```diff
@@ -136,27 +229,82 @@ function NoTokensState({ message }: { message: ReactNode }) {
   )
 }
 
-const LOADING_ROWS = Array.from({ length: 100 })
-  .fill(0)
-  .map((_item, index) => <LoadingRow key={index} />)
-
 export function LoadingTokenTable() {
   return (
-    <GridContainer>
+    <GridContainer fixedHeight={true}>
       <HeaderRow />
-      <TokenRowsContainer>{LOADING_ROWS}</TokenRowsContainer>
+      <TokenDataContainer>
+        <AutoSizer>
+          {({ height, width }) => (
+            <FixedSizeList className="List" height={height} width={width} itemCount={MAX_TOKENS_TO_LOAD} itemSize={70}>
+              {({ index, style }) => <LoadingRow style={style} key={index} />}
+            </FixedSizeList>
+          )}
+        </AutoSizer>
+      </TokenDataContainer>
     </GridContainer>
   )
 }
 
-export default function TokenTable({ data }: { data: TokenData[] | undefined }) {
+interface TokenRowProps {
+  data: TokenData[]
+  index: number
+  style: CSSProperties
+}
+
+export default function TokenTable() {
   const showFavorites = useAtomValue<boolean>(showFavoritesAtom)
   const timePeriod = useAtomValue<TimePeriod>(filterTimeAtom)
-  const filteredTokens = useFilteredTokens(data)
-  const sortedFilteredTokens = useSortedTokens(filteredTokens)
+  const filterString = useAtomValue(filterStringAtom)
+  const sortCategory = useAtomValue(sortCategoryAtom)
+  const sortDirection = useAtomValue(sortDirectionAtom)
+
+  const [page, setPage] = useState<number>(1)
+  const [topTokens, setTopTokens] = useState<TokenData[]>(
+    Array.from({ length: MAX_TOKENS_TO_LOAD }).map((_) => ({ loading: true } as TokenData))
+  )
+  const [error, setError] = useState<any>(undefined)
+
+  const preloadedTopTokens = useTopTokenQuery(MAX_TOKENS_TO_LOAD, 1) ?? []
+  const nonDefaultFiltersExist = hasNonDefaultFilters(filterString, showFavorites)
+  const nonDefaultSortingParamsExist = hasNonDefaultSortingParams(sortCategory, sortDirection, timePeriod)
+  const usePreloadedTokens = nonDefaultFiltersExist || nonDefaultSortingParamsExist
```

Comment:
`exist` here is a little confusing to me here. Its usage as a flag to skip sort/filtering is also kinda confusing to me.